### ISSUE TITLE
Cellpose segmentation on GPU or CPU produces kernel crash without traceback on large 2D ome.tiff files

### ISSUE THREAD

**Issue by gargerd** [MouseLand/cellpose#814]
Hi, 

I am trying to run a Cellpose segmentation on large ome.tiff files (dimensions are ~ 36000*25000 pixels), but my kernel keeps crashing at the ```[INFO] ~~~ FINDING MASKS ~~~``` step without any errors or traceback. The script runs without issues on a 640 * 480 .tif image I downloaded from the internet, so my suspicion is an out-of-memory error (possibly related to issues #520 and #591), but none of the workarounds proposed in those issue helped me (setting `batch_size=1`  or setting `flow_threshold=0`), also I don't get any out-of-memory errors. My suspicion is underpinned by the fact that the less RAM I am using, the earlier the kernel crash happens. 

I run this script on a SLURM based HPC cluster, and in order to test out a memory bottleneck I even ran the script using 2 NVIDIA Tesla A100 GPUs (2x80 GB RAM) but the kernel still crashed, which should not be the case as the image is only 245 MBs in size. 

I also tried to run my script using only CPUs (6 cores + 128 GB memory), but it got stuck at the ```[INFO] ~~~ FINDING MASKS ~~~``` step, and after 3 hours and I stopped the script. I did not get a Kernel crash in this case, although it could be that I didn't wait long enough, as Cellpose is way slower on CPU compared to GPU.

The crash happens in Jupyter notebook and also when running Cellpose from the command line.

Unfortunately I cannot share the ome.tiff image as it is sensitive information, but __here is the relevant Jupyter code I am using__:

```
import skimage.io
import numpy as np
import os
from cellpose import models,core
import logging
from cellpose import io 
logger = io.logger_setup()
from imageio.v2 import imread


fn='../slides/morphology_mip.ome.tif'
img=imread(fn)

use_GPU = core.use_gpu()
model = models.Cellpose(gpu=use_GPU,model_type='cyto')
hyperparams={"batch_size":1,"channel_axis":None,"z_axis":None,"invert":False,"normalize":True,
            "diameter":30.0,"do_3D":False,"anisotropy":None,"net_avg":False,"augment":False,"tile":True,
            "tile_overlap":0.1,"resample":True,"interp":True,"flow_threshold":0.0,"cellprob_threshold":0.0,
            "min_size":15,"stitch_threshold":0.0,"rescale":False,"progress":False,"model_loaded":False}

res, _, _, _ = model.eval(
            img,
            channels=[0, 0],
            **hyperparams)
```

__Jupyter output__:
```
Input image shape: (34056, 28436)
2023-11-05 19:00:29,179 [INFO] ** TORCH CUDA version installed and working. **
2023-11-05 19:00:29,180 [INFO] ** TORCH CUDA version installed and working. **
2023-11-05 19:00:29,181 [INFO] >>>> using GPU
2023-11-05 19:00:29,182 [INFO] >> cyto << model set to be used
2023-11-05 19:00:29,650 [INFO] >>>> model diam_mean =  30.000 (ROIs rescaled to this size during training)
2023-11-05 19:00:29,654 [INFO] ~~~ FINDING MASKS ~~~
Canceled future for execute_request message before replies were done
The Kernel crashed while executing code in the the current cell or a previous cell. Please review the code in the cell(s) to identify a possible cause of the failure. Click [here](https://aka.ms/vscodeJupyterKernelCrash) for more info. View Jupyter [log](command:jupyter.viewOutput) for further details.
```
__CLI command__:
`python -m cellpose --use_gpu --verbose --image_path $input_fn`

__Terminal output when using CLI command__:
```
2023-11-05 19:07:30,529 [INFO] WRITING LOG OUTPUT TO /home/gargerd/.cellpose/run.log
2023-11-05 19:07:32,107 [INFO] ** TORCH CUDA version installed and working. **
2023-11-05 19:07:32,108 [INFO] >>>> using GPU
2023-11-05 19:07:32,109 [INFO] >>>> running cellpose on 1 images using chan_to_seg GRAY and chan (opt) NONE
2023-11-05 19:07:32,109 [INFO] ** TORCH CUDA version installed and working. **
2023-11-05 19:07:32,109 [INFO] >>>> using GPU
2023-11-05 19:07:32,109 [INFO] >> cyto << model set to be used
2023-11-05 19:07:32,421 [INFO] >>>> model diam_mean =  30.000 (ROIs rescaled to this size during training)
2023-11-05 19:07:32,424 [INFO] >>>> using diameter 30.000 for all images
2023-11-05 19:07:32,426 [INFO] 0%|          | 0/1 [00:00<?, ?it/s]
2023-11-05 19:07:41,322 [INFO] ~~~ FINDING MASKS ~~~
run_cell_segm_benchm.sh: line 84: 290302 Killed                  python -m cellpose --use_gpu --verbose --image_path $input_fn
```
Any help would be appreciated!

__I am using python 3.10.0, here are the packages in my environment:__

Package                       Version
----------------------------- --------------------
alabaster                     0.7.12
anaconda-client               1.11.0
anaconda-navigator            2.3.1
anaconda-project              0.11.1
anyio                         3.5.0
appdirs                       1.4.4
argon2-cffi                   21.3.0
argon2-cffi-bindings          21.2.0
arrow                         1.2.2
astroid                       2.11.7
astropy                       5.1
atomicwrites                  1.4.0
attrs                         21.4.0
Automat                       20.2.0
autopep8                      1.6.0
Babel                         2.9.1
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
backports.tempfile            1.0
backports.weakref             1.0.post1
bcrypt                        3.2.0
beautifulsoup4                4.11.1
binaryornot                   0.4.4
bitarray                      2.5.1
bkcharts                      0.2
black                         22.6.0
bleach                        4.1.0
bokeh                         2.4.3
boto3                         1.24.28
botocore                      1.27.28
Bottleneck                    1.3.5
brotlipy                      0.7.0
certifi                       2022.9.14
cffi                          1.15.1
chardet                       4.0.0
charset-normalizer            2.0.4
click                         8.0.4
cloudpickle                   2.0.0
clyent                        1.2.2
colorama                      0.4.5
colorcet                      3.0.0
conda                         22.9.0
conda-build                   3.22.0
conda-content-trust           0.1.3
conda-pack                    0.6.0
conda-package-handling        1.9.0
conda-repo-cli                1.0.20
conda-token                   0.4.0
conda-verify                  3.4.2
constantly                    15.1.0
cookiecutter                  1.7.3
cryptography                  37.0.1
cssselect                     1.1.0
cycler                        0.11.0
Cython                        0.29.32
cytoolz                       0.11.0
daal4py                       2021.6.0
dask                          2022.7.0
datashader                    0.14.1
datashape                     0.5.4
debugpy                       1.5.1
decorator                     5.1.1
defusedxml                    0.7.1
diff-match-patch              20200713
dill                          0.3.4
distributed                   2022.7.0
docutils                      0.18.1
entrypoints                   0.4
et-xmlfile                    1.1.0
fastjsonschema                2.16.2
filelock                      3.6.0
flake8                        4.0.1
Flask                         1.1.2
fonttools                     4.25.0
fsspec                        2022.7.1
future                        0.18.2
gensim                        4.1.2
glob2                         0.7
gmpy2                         2.1.2
greenlet                      1.1.1
h5py                          3.7.0
HeapDict                      1.0.1
holoviews                     1.15.0
hvplot                        0.8.0
hyperlink                     21.0.0
idna                          3.3
imagecodecs                   2021.8.26
imageio                       2.19.3
imagesize                     1.4.1
importlib-metadata            4.11.3
incremental                   21.3.0
inflection                    0.5.1
iniconfig                     1.1.1
intake                        0.6.5
intervaltree                  3.1.0
ipykernel                     6.15.2
ipython                       7.31.1
ipython-genutils              0.2.0
ipywidgets                    7.6.5
isort                         5.9.3
itemadapter                   0.3.0
itemloaders                   1.0.4
itsdangerous                  2.0.1
jdcal                         1.4.1
jedi                          0.18.1
jeepney                       0.7.1
jellyfish                     0.9.0
Jinja2                        2.11.3
jinja2-time                   0.2.0
jmespath                      0.10.0
joblib                        1.1.0
json5                         0.9.6
jsonschema                    4.16.0
jupyter                       1.0.0
jupyter_client                7.3.4
jupyter-console               6.4.3
jupyter_core                  4.11.1
jupyter-server                1.18.1
jupyterlab                    3.4.4
jupyterlab-pygments           0.1.2
jupyterlab-server             2.10.3
jupyterlab-widgets            1.0.0
keyring                       23.4.0
kiwisolver                    1.4.2
lazy-object-proxy             1.6.0
libarchive-c                  2.9
llvmlite                      0.38.0
locket                        1.0.0
lxml                          4.9.1
lz4                           3.1.3
Markdown                      3.3.4
MarkupSafe                    2.0.1
matplotlib                    3.5.2
matplotlib-inline             0.1.6
mccabe                        0.7.0
mistune                       0.8.4
mkl-fft                       1.3.1
mkl-random                    1.2.2
mkl-service                   2.4.0
mock                          4.0.3
mpmath                        1.2.1
msgpack                       1.0.3
multipledispatch              0.6.0
munkres                       1.1.4
mypy-extensions               0.4.3
navigator-updater             0.3.0
nbclassic                     0.3.5
nbclient                      0.5.13
nbconvert                     6.4.4
nbformat                      5.5.0
nest-asyncio                  1.5.5
networkx                      2.8.4
nltk                          3.7
nose                          1.3.7
notebook                      6.4.12
numba                         0.55.1
numexpr                       2.8.3
numpy                         1.21.5
numpydoc                      1.4.0
olefile                       0.46
openpyxl                      3.0.10
packaging                     21.3
pandas                        1.4.4
pandocfilters                 1.5.0
panel                         0.13.1
param                         1.12.0
parsel                        1.6.0
parso                         0.8.3
partd                         1.2.0
pathlib                       1.0.1
pathspec                      0.9.0
patsy                         0.5.2
pep8                          1.7.1
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        9.2.0
pip                           22.2.2
pkginfo                       1.8.2
platformdirs                  2.5.2
plotly                        5.9.0
pluggy                        1.0.0
ply                           3.11
poyo                          0.5.0
prometheus-client             0.14.1
prompt-toolkit                3.0.20
Protego                       0.1.16
psutil                        5.9.0
ptyprocess                    0.7.0
py                            1.11.0
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pycodestyle                   2.8.0
pycosat                       0.6.3
pycparser                     2.21
pyct                          0.4.8
pycurl                        7.45.1
PyDispatcher                  2.0.5
pydocstyle                    6.1.1
pyerfa                        2.0.0
pyflakes                      2.4.0
Pygments                      2.11.2
PyHamcrest                    2.0.2
PyJWT                         2.4.0
pylint                        2.14.5
pyls-spyder                   0.4.0
pyodbc                        4.0.34
pyOpenSSL                     22.0.0
pyparsing                     3.0.9
PyQt5-sip                     12.11.0
pyrsistent                    0.18.0
PySocks                       1.7.1
pytest                        7.1.2
python-dateutil               2.8.2
python-lsp-black              1.2.1
python-lsp-jsonrpc            1.0.0
python-lsp-server             1.5.0
python-slugify                5.0.2
python-snappy                 0.6.0
pytz                          2022.1
pyviz-comms                   2.0.2
PyWavelets                    1.3.0
pyxdg                         0.27
PyYAML                        6.0
pyzmq                         23.2.0
QDarkStyle                    3.0.2
qstylizer                     0.1.10
QtAwesome                     1.0.3
qtconsole                     5.3.2
QtPy                          2.2.0
queuelib                      1.5.0
regex                         2022.7.9
requests                      2.28.1
requests-file                 1.5.1
rope                          0.22.0
Rtree                         0.9.7
ruamel-yaml-conda             0.15.100
s3transfer                    0.6.0
scikit-image                  0.19.2
scikit-learn                  1.0.2
scikit-learn-intelex          2021.20221004.171807
scipy                         1.9.1
Scrapy                        2.6.2
seaborn                       0.11.2
SecretStorage                 3.3.1
Send2Trash                    1.8.0
service-identity              18.1.0
setuptools                    63.4.1
sip                           6.6.2
six                           1.16.0
smart-open                    5.2.1
sniffio                       1.2.0
snowballstemmer               2.2.0
sortedcollections             2.1.0
sortedcontainers              2.4.0
soupsieve                     2.3.1
Sphinx                        5.0.2
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
spyder                        5.3.3
spyder-kernels                2.3.3
SQLAlchemy                    1.4.39
statsmodels                   0.13.2
sympy                         1.10.1
tables                        3.6.1
tabulate                      0.8.10
TBB                           0.2
tblib                         1.7.0
tenacity                      8.0.1
terminado                     0.13.1
testpath                      0.6.0
text-unidecode                1.3
textdistance                  4.2.1
threadpoolctl                 2.2.0
three-merge                   0.1.1
tifffile                      2021.7.2
tinycss                       0.4
tldextract                    3.2.0
toml                          0.10.2
tomli                         2.0.1
tomlkit                       0.11.1
toolz                         0.11.2
tornado                       6.1
tqdm                          4.64.1
traitlets                     5.1.1
Twisted                       22.2.0
typing_extensions             4.3.0
ujson                         5.4.0
Unidecode                     1.2.0
urllib3                       1.26.11
w3lib                         1.21.0
watchdog                      2.1.6
wcwidth                       0.2.5
webencodings                  0.5.1
websocket-client              0.58.0
Werkzeug                      2.0.3
whatthepatch                  1.0.2
wheel                         0.37.1
widgetsnbextension            3.5.2
wrapt                         1.14.1
wurlitzer                     3.0.2
xarray                        0.20.1
xlrd                          2.0.1
XlsxWriter                    3.0.3
yapf                          0.31.0
zict                          2.1.0
zipp                          3.8.0
zope.interface                5.4.0

**Comment by mrariden** [comment]
Cellpose needs a lot of memory, and it scales with the area of the image. There are also some bugs with memory usage. 

Can you try to use this branch: https://github.com/mrariden/cellpose/tree/speedy_improvements ? You can use the following commands to install it to your conda env if your have git available: 
```bash
conda activate <your env>
pip install git+https://github.com/mrariden/cellpose.git@speedy_improvements
```
Then try to use cellpose again and share your results. These changes are not yet included in the main branch.

**Comment by gargerd** [comment]
Thank you @mrariden for your suggestion!

I tried to re-run it with the speedy_improvements branch, but the same issue happened. 

Cellpose obviously needs a lot of memory, but don't you think 2GPUs with 160GB memory should suffice for a 245MB ome.tiff file with (34056x28436) pixels? Do you have any rule-of-thumb or ballpark value about the memory requirements regarding image input size?

Also, could a wrong image format/parameter settings cause this issue? My image array is not in an RGB format, it has the shape of (34056x28436) and its values range from 0-~7500, corresponding to fluorescence intensity measured in one channel. Is the parameter `channels=[0,0]` appropriate for this kind of input?

I just want to rule out any other minor and obvious issue before moving on the request large resources on the HPC.

**Comment by gargerd** [comment]
@mrariden: I tried again with more GPUs and memory and this time the run was successful with the speedy_improvements branch, so the solution is more manpower:)

Thanks for the help!
